crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -457,6 +527,20 @@ impl<'db> SemanticIndexBuilder<'db> {
     /// Record an expression that needs to be a Salsa ingredient, because we need to infer its type
     /// standalone (type narrowing tests, RHS of an assignment.)
     fn add_standalone_expression(&mut self, expression_node: &ast::Expr) -> Expression<'db> {
+        self.add_standalone_expression_impl(expression_node, false)
+    }
```

Comment:
I wonder if we could use this enum here (or do something like it) for the second parameter of `add_standalone_expression_impl`, rather than using a `bool`? We could move it (and rename it -- `ValueOrTypeExpression`?).

https://github.com/astral-sh/ruff/blob/fab86de3ef91460e58496f74ba1ce3edc4b9d69c/crates/red_knot_python_semantic/src/types.rs#L3701-L3711

I find it hard to tell why a `true` or `false` argument is being passed into a function in Rust without jumping to the definition of the function in question (which is hard if I'm reading a diff on GitHub).

Reply:
> I find it hard to tell why a true or false argument is being passed into a function in Rust without jumping to the definition of the function in question

... which is why I created two variants of this function that call ..._impl, so you don't have to see it in the "actual" code :-).

But I'm also fine with an enum, if you think that's clearer.

Reply:
heh, yeah. To give the full story here: while reading through the diff, I persuaded myself that it wasn't worth making this nit comment (for the reason you give), carried on reading the diff... and then saw that you were also adding an `infer_as_type_expression: bool` field to the `Expression` struct, which made me wonder if an enum like this could be more generally useful :-)